Makefile

```
@@ -60,17 +60,24 @@ exec: estimate sipnet transpose subsetData histutil
 # with the default set to 'sipnet', we can have all do (close to) everything
 all: exec document
 
-# Only update docs if source files or Doxyfile have changed
-document: .doxygen.stamp
-
-# Reminder: this is the implicit build command
-# %.o: %.c
-# 	$(CC) $(CFLAGS) -c $< -o $@
-
-.doxygen.stamp: $(CFILES) $(DOXYFILE)
-	@echo "Running Doxygen..."
-	doxygen $(DOXYFILE)
-	@touch .doxygen.stamp
+# Build documentation with both Doxygen and Mkdocs
+document: .doxygen.stamp .mkdocs.stamp
+	@echo "Documentation updated."
+
+.doxygen.stamp: $(DOXYFILE) $(CFILES)
+	@if [ ! -d $(DOXYGEN_HTML_DIR) ] || [ $(DOXYFILE) -nt .doxygen.stamp ] || \
+	   find $(CFILES) -newer .doxygen.stamp | read dummy; then \
+		echo "Running Doxygen..."; \
+		doxygen $(DOXYFILE); \
+		touch .doxygen.stamp; \
+	else \
+		echo "Doxygen is up-to-date."; \
+	fi
+
+.mkdocs.stamp:
```

If we want to make sure doxygen goes first, have this target depend on `.doxygen.stamp`

---

yes, due to this firstly *doxygen* files are downloaded and then *mkdocs* files are downloaded.

---

do you mean 'compiled' as opposed to downloaded?

I think @Alomir is suggesting
```suggestion
.mkdocs.stamp: .doxygen.stamp
```